pkg/fs/remote/azure/azblob.go

```diff
@@ -0,0 +1,281 @@
+		}
+		containerName = parts[0]
+		cfg.Azure.Container = containerName
+		if len(parts) == 2 {
+			rawPath = parts[1]
+		} else {
+			rawPath = ""
+		}
+	} else {
+		// Remove duplicated container prefix if caller already passed it via the path.
+		prefix := containerName + "/"
+		// Always attempt to trim the container prefix; TrimPrefix is safe even if
+		// rawPath does not start with the prefix.
+		rawPath = strings.TrimPrefix(rawPath, prefix)
+	}
+
+	// Ensure basePath does not start with a leading slash to avoid blobs having
+	// an extra leading segment and to make listing logic consistent.
+	basePath := strings.TrimPrefix(rawPath, "/")
+
+	client, err := buildClient(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	verifier, err := checksum.DefaultSHA256Verifier()
+	if err != nil {
+		return nil, err
+	}
+
+	// ensure container exists (create if not)
+	if err := ensureContainer(context.Background(), client, containerName); err != nil {
+		return nil, err
+	}
+
+	return &blobFS{
+		client:    client,
+		container: containerName,
+		basePath:  basePath,
+		verifier:  verifier,
+	}, nil
+}
+
+func buildClient(cfg *config.FsConfig) (*azblob.Client, error) {
+	azureCfg := cfg.Azure
+	endpoint := azureCfg.AzureEndpoint
+	if connStr := os.Getenv("AZURE_STORAGE_CONNECTION_STRING"); connStr != "" {
+		return azblob.NewClientFromConnectionString(connStr, nil)
+	}
+
+	// Prefer SAS token if supplied
+	if azureCfg.AzureSASToken != "" {
+		// Ensure the token starts without leading "?"
+		sas := strings.TrimPrefix(azureCfg.AzureSASToken, "?")
+		var serviceURL string
+		if strings.Contains(endpoint, "?") {
+			serviceURL = endpoint + "&" + sas
+		} else {
+			serviceURL = endpoint + "?" + sas
+		}
+		return azblob.NewClientWithNoCredential(serviceURL, nil)
+	}
+
+	if azureCfg.AzureAccountName != "" && azureCfg.AzureAccountKey != "" {
+		endpoint := azureCfg.AzureEndpoint
+		if endpoint == "" {
+			endpoint = fmt.Sprintf("https://%s.blob.core.windows.net/", azureCfg.AzureAccountName)
+		}
+		cred, err := azblob.NewSharedKeyCredential(
+			azureCfg.AzureAccountName,
+			azureCfg.AzureAccountKey,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("invalid shared key credential: %w", err)
+		}
+		return azblob.NewClientWithSharedKeyCredential(endpoint, cred, nil)
+	}
+
+	return nil, errors.New("no Azure credentials found: set AZURE_STORAGE_CONNECTION_STRING, sas_token, or account_name+account_key")
+}
+
+func ensureContainer(ctx context.Context, client *azblob.Client, containerName string) error {
+	_, err := client.CreateContainer(ctx, containerName, nil)
+	if err == nil {
+		return nil
+	}
+	var respErr *azcore.ResponseError
+	if errors.As(err, &respErr) {
+		if respErr.ErrorCode == "ContainerAlreadyExists" {
+			return nil
+		}
+	}
+	return fmt.Errorf("failed to ensure container %s exists: %w", containerName, err)
+}
+
+func (b *blobFS) getFullPath(p string) string {
+	if b.basePath == "" {
+		return p
+	}
+	return path.Join(b.basePath, p)
```

Comment:
The function uses `path.Join()` which is platform-specific. For cloud storage, it should handle path separators explicitly for cross-platform compatibility.